compiler/src/dotty/tools/dotc/core/NameKinds.scala

```diff
@@ -61,8 +61,7 @@ object NameKinds {
     def infoString: String
   }
 
-  /** The kind of SimpleNames */
-  val SimpleNameKind = new NameKind(UTF8) { self =>
+  object SimpleNameKind extends NameKind(UTF8) { self =>
```

Comment:
But this is another issue. The problem was not threading but the fact that the registration was  not performed at all, since objects are lazy.